src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -593,99 +527,165 @@ function flux_lax_friedrichs(u_ll, u_rr, orientation, equations::CompressibleEul
   f_rr = calcflux(u_rr, orientation, equations)
 
   λ_max = max(v_mag_ll, v_mag_rr) + max(c_ll, c_rr)
-  f1 = 1/2 * (f_ll[1] + f_rr[1]) - 1/2 * λ_max * (rho1_rr   - rho1_ll)
-  f2 = 1/2 * (f_ll[2] + f_rr[2]) - 1/2 * λ_max * (rho2_rr   - rho2_ll)
-  f3 = 1/2 * (f_ll[3] + f_rr[3]) - 1/2 * λ_max * (rho_v1_rr - rho_v1_ll)
-  f4 = 1/2 * (f_ll[4] + f_rr[4]) - 1/2 * λ_max * (rho_v2_rr - rho_v2_ll)
-  f5 = 1/2 * (f_ll[5] + f_rr[5]) - 1/2 * λ_max * (rho_e_rr  - rho_e_ll)
+  
+  f  = SVector{length(gamma)+3,Float64}(1/2 * (f_ll[i] + f_rr[i]) - 1/2 * λ_max * (u_rr[i] - u_ll[i]) for i = 1:length(gamma)+3)
 
-  return SVector(f1, f2, f3, f4, f5)
+  return f
 end
 
 
 @inline function max_abs_speeds(u, equations::CompressibleEulerMulticomponentEquations2D)
-  @unpack cv1, cv2, gamma1, gamma2 = equations
-  rho1, rho2, rho_v1, rho_v2, rho_e = u
+  @unpack cv, gamma = equations
+  rho_v1, rho_v2, rho_e = u
+  rho   = 0
+
+  for i = 1:length(gamma)
+    rho += u[i+3]
+  end  
 
-  rho   = rho1 + rho2
   v1    = rho_v1 / rho
   v2    = rho_v2 / rho
 
-  gamma = (rho1*cv1*gamma1 + rho2*cv2*gamma2)/(rho1*cv1 + rho2*cv2)
-  p     = (gamma - 1) * (rho_e - 1/2 * rho * (v1^2 + v2^2))
-  c     = sqrt(gamma * p / rho)
+  gammas= getgamma(u, equations)
```

Comment:
@sloede should chime in and say whether he would prefer `gamma(equations)` or `getgamma(equations)`. But we don't need `u` as argument, do we? Such "getter methods" often do not use `get` in their name in Julia.

Reply:
I changed it to "totalgamma" since "gamma" collides with the unpacked gamma. u is needed to get the density fractions of the species to calculate the total gamma.

done in 4c720b0